packages/fhir-team-management/src/utils.ts

```diff
@@ -112,40 +120,95 @@ export const updatePractitionerRoles = (
     const organizationId = orgId as string;
     const role =
       existingRolesByOrgPractIds[`${organizationResourceType}/${organizationId}`][practId];
-    // invariant if we are removing a practitioner initially assigned to this team
-    // then a role should exist
-    return () => serve.delete(role.id as string);
+    // remove organization assignment from role
+    const { organization, ...rest } = role;
+    return () => serve.update(rest);
   });
 
-  // TODO - possibility of posting this as a bundle in a single api call
-  const additionPromises = toAdd
-    .map((practId) => {
-      const practitioner = practitionersById[practId];
-      const name = getObjLike(practitioner.name, 'use', HumanNameUseCodes.OFFICIAL, true)[0];
-      const practitionerRole: IPractitionerRole = {
+  const practitionerRolesModifyPromises = [];
+
+  if (toAdd.length > 0) {
+    const organizationPayload = {
+      reference: `${organizationResourceType}/${orgId}`,
+      display: organization.name,
+    };
+
+    // get existing practitioner roles for all new practitioners to be added to organization
+    const existingPractitionerRoles = (await loadAllResources(
+      baseUrl,
+      practitionerRoleResourceType,
+      {
+        practitioner: toAdd.join(),
+      }
+    ).then((resp) => getResourcesFromBundle(resp))) as IPractitionerRole[];
+
+    // inject organization attribute into existing practitioner roles
+    const existingPractitionerRolesPromises = existingPractitionerRoles
+      .map((practitionerRole) => ({
+        ...practitionerRole,
+        organization: organizationPayload,
+      }))
+      .map((practitionerRole) => {
+        return () => serve.update(practitionerRole);
+      });
+
+    practitionerRolesModifyPromises.push(...existingPractitionerRolesPromises);
+
+    // create practitioner roles against practitioners to be added if they don't exist
+    const practitionersWithExistingPractitionerRoles = existingPractitionerRoles.map(
+      (existingPractitionerRole) => existingPractitionerRole.practitioner?.reference
+    );
+
+    const practitionersWithoutExistingPractitionerRoles = toAdd.filter(
+      (practitionerId) => !practitionersWithExistingPractitionerRoles.includes(practitionerId)
+    );
+
+    for (const practitionerID of practitionersWithoutExistingPractitionerRoles) {
+      const newPractitionerRoleResourceID = v4();
+      const practitioner = practitionersById[practitionerID];
+      // get secondary identifier (keycloak uuid) from practitioner
+      const practitionerSecondaryIdentifier = practitioner.identifier?.find(
+        (identifier) => identifier.use === 'secondary'
+      );
+      const practitionerDisplayName = getObjLike(
+        practitioner.name,
+        'use',
+        HumanNameUseCodes.OFFICIAL,
+        true
```

Comment:
LG, just curious if this logic here is exercised in a test.

Reply:
@p-netm yes. both flows are covered in the added mocks (existing and non-existent roles)

Reply:
ok, cool.